src/chains.jl

```diff
@@ -757,10 +756,24 @@ function _cat(::Val{1}, c1::Chains, args::Chains...)
                       iter = mapreduce(range, vcat, args; init=range(c1)),
                       var = nms,
                       chain = chns)
+    weights = _weights_cat(c1.weights, getproperty.(args, :weights)...)
```

Comment:
If possible, avoid splatting but use something similar to `mapreduce`. Maybe use
```suggestion
    weights = _weights_cat(c1, args)
```
and handle the weight extraction in `_weights_cat` (you can still dispatch on weight types by dispatching on the type parameter).

Reply:
Can you address this comment and avoid splatting?